include/translated-term.php

```diff
@@ -338,4 +338,81 @@ protected function get_db_infos() {
 			'default_alias' => 't',
 		);
 	}
+
+	/**
+	 * Wrapper of `wp_insert_term` with the language.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string       $term     The term name to add.
+	 * @param string       $taxonomy The taxonomy to which to add the term.
+	 * @param PLL_Language $language The term language.
+	 * @param array|string $args     The taxonomy of the term.
+	 * @return array|WP_Error An array of the new term, WP_Error otherwise.
+	 *
+	 * @phpstan-param array{
+	 *   alias_of?: string,
+	 *   description?: string,
+	 *   parent?: int,
+	 *   slug?: string,
+	 * } $args
+	 */
+	public function insert_term( string $term, string $taxonomy, PLL_Language $language, $args = array() ) {
+		$set_language_for_term_slug = function () use ( $language ) {
+			return $language;
+		};
+
+		add_filter( 'pll_inserted_term_language', $set_language_for_term_slug, 20 ); // After Polylang's filter.
+		$term = wp_insert_term( $term, $taxonomy, $args );
+		remove_filter( 'pll_inserted_term_language', $set_language_for_term_slug, 20 );
+
+		if ( is_wp_error( $term ) ) {
+			// Something went wrong!
+			return $term;
+		}
+
+		$term_id = (int) $term['term_id'];
+		$this->model->term->set_language( $term_id, $language );
+
+		return $term;
+	}
+
+	/**
+	 * Wrapper of `wp_update_term` with the language.
+	 *
+	 * @since 3.7
+	 *
+	 * @param int          $term_id  The ID of the term.
+	 * @param string       $taxonomy The taxonomy of the term.
+	 * @param PLL_Language $language The term language.
+	 * @param array        $args     The taxonomy of the term.
+	 * @return array|WP_Error An array containing the term_id and term_taxonomy_id, WP_Error otherwise.
+	 */
+	public function update_term( int $term_id, string $taxonomy, PLL_Language $language, array $args = array() ) {
+		$set_language_for_term_slug = function () use ( $language ) {
+			return $language;
+		};
+
+		$term = get_term( $term_id );
+		if ( ! $term instanceof WP_Term ) {
+			return new WP_Error( 'invalid_term', __( 'Empty Term.', 'polylang' ) );
+		}
+
+		$term_parent              = $term->parent;
+		$get_inserted_term_parent = function () use ( $term_parent ) {
+			return $term_parent;
+		};
+
+		// Set term parent and language for suffixed slugs.
+		add_filter( 'pll_inserted_term_language', $set_language_for_term_slug, 20 ); // After Polylang's filter.
```

Comment:
Why do we have to come after Polylang filter? The other callback doesn't check there is already a language added?

Reply:
Actually I don't remember why we have this priority. 

It seems to me that the other callback indeed check if there is already a language and returns it if this is the case.

Reply:
So we could use the default priority, right?

Reply:
Yep, it's done.